### ISSUE TITLE
Unable to save set data

### ISSUE THREAD

**Issue by jp-lesage** [brandonp2412/Flexify#17]
Hi!

After all the issues with updating from v.1.0.44 to the latest version, set data does not save when tapping the save button.

The set counter does not increment, and no record gets generated for the graphs. I tried with the built-in plans and the one I imported and both have the same issue. There's no error message displayed on screen.

I had to revert back to v.1.0.44 to be able to save anything. Uninstall/clean cache/reinstall does not help.

Thanks!

**Comment by jp-lesage** [comment]
Hi!

I did some more testing and it seems the issue started in v1.0.47. I tried every subsequent version (by uninstalling, reinstalling, and using the built-in data) and they all behave the same way: the save button does nothing when adding a new set.

After more digging, sets will not be added/saved if rest timers are disabled in the application settings. Since I don't use them, it's blocking me from adding sets... I'll temporarily enable them to work around that issue.

I took a quick look at the code, but I'm not familiar enough with Dart to offer anything other than wild guesses. I believe that specific view is defined in `lib/start_plan_page.dart`, so either the `onPressed` event handler of the save button isn't called, or the call to `save()` silently fails. I didn't see any change that would cause this, beside one extra condition being checked on line 158 (the check on `mounted`).

Thanks!
